optuna/samplers/_search_space/intersection.py

```diff
@@ -59,20 +59,20 @@ def calculate(self, study: Study, ordered_dict: bool = False) -> Dict[str, BaseD
             if self._study_id != study._study_id:
                 raise ValueError("`IntersectionSearchSpace` cannot handle multiple studies.")
 
-        states_of_interest = [optuna.trial.TrialState.COMPLETE]
+        states_of_interest = [optuna.trial.TrialState.COMPLETE, optuna.trial.TrialState.RUNNING]
```

Comment:
```suggestion
        states_of_interest = [optuna.trial.TrialState.COMPLETE, optuna.trial.TrialState.WAITING, optuna.trial.TrialState.RUNNING]
```

Reply:
Thanks!

Reply:
Could you add unit tests to catch this error? 

Currently, we do not check that `trials in the state of RUNNING, WAITING are not counted, but later they are counted after completion` in unit tets. 
Only we checked that FAIL and PRUNED are skipped here.
https://github.com/optuna/optuna/blob/master/tests/samplers_tests/search_space_tests/test_intersection.py#L54-L55

Reply:
@contramundum53 Sorry, I added a comment on the resolved conversation. Could you check it
?

Reply:
@keisuke-umezawa I added a test case for waiting trial. PTAL.